examples/simple-chat/src/simple_chat.ts

```diff
@@ -27,51 +27,91 @@ class ChatUI {
+   */
+  public static CreateAsync = async (chat: ChatInterface, localChat: ChatInterface) => {
+    const chatUI = new ChatUI();
     // use web worker to run chat generation in background
-    this.chat = chat;
-    this.localChat = localChat;
+    chatUI.chat = chat;
+    chatUI.localChat = localChat;
     // get the elements
-    this.uiChat = getElementAndCheck("chatui-chat");
-    this.uiChatInput = getElementAndCheck("chatui-input") as HTMLInputElement;
-    this.uiChatInfoLabel = getElementAndCheck("chatui-info-label") as HTMLLabelElement;
+    chatUI.uiChat = getElementAndCheck("chatui-chat");
+    chatUI.uiChatInput = getElementAndCheck("chatui-input") as HTMLInputElement;
+    chatUI.uiChatInfoLabel = getElementAndCheck("chatui-info-label") as HTMLLabelElement;
     // register event handlers
     getElementAndCheck("chatui-reset-btn").onclick = () => {
-      this.onReset();
+      chatUI.onReset();
     };
     getElementAndCheck("chatui-send-btn").onclick = () => {
-      this.onGenerate();
+      chatUI.onGenerate();
     };
     // TODO: find other alternative triggers
     getElementAndCheck("chatui-input").onkeypress = (event) => {
       if (event.keyCode === 13) {
-        this.onGenerate();
+        chatUI.onGenerate();
       }
     };
 
+    // When we detect low maxStorageBufferBindingSize, we assume that the device (e.g. an Android
+    // phone) can only handle small models and make all other models unselectable. Otherwise, the
+    // browser may crash. See https://github.com/mlc-ai/web-llm/issues/209.
+    let restrictModels = false;
+    const lowRequirementModels = new Set<string>([
+      'RedPajama-INCITE-Chat-3B-v1-q4f16_1-1k',
+      'RedPajama-INCITE-Chat-3B-v1-q4f32_1-1k'
+    ])
+    let maxStorageBufferBindingSize: number;
+    try {
+      maxStorageBufferBindingSize = await chat.getMaxStorageBufferBindingSize();
+    } catch (err) {
+      chatUI.appendMessage("error", "Init error, " + err.toString());
+      console.log(err.stack);
+      return;
+    }
+    const androidMaxStorageBufferBindingSize = 1 << 27;  // 128MB
+    if (maxStorageBufferBindingSize <= androidMaxStorageBufferBindingSize) {
+      chatUI.appendMessage("init", "Your device seems to have " +
+        "limited resources, so we restrict the selectable models.");
+      restrictModels = true;
+    }
+
+    // Populate modelSelector
     const modelSelector = getElementAndCheck("chatui-select") as HTMLSelectElement;
-    for (let i = 0; i < this.config.model_list.length; ++i) {
-      const item = this.config.model_list[i];
+    for (let i = 0; i < chatUI.config.model_list.length; ++i) {
+      const item = chatUI.config.model_list[i];
       const opt = document.createElement("option");
       opt.value = item.local_id;
       opt.innerHTML = item.local_id;
       opt.selected = (i == 0);
+      if (restrictModels && !lowRequirementModels.has(item.local_id)) {
+        opt.disabled = true;
```

Comment:
May I recommend a way to bypass this from a URL argument for instance so that developers who are investigating in crashes can reproduce?

```suggestion
        const params = new URLSearchParams(location.search);
        opt.disabled = !params.has("bypassRestrictions");
  ```

Reply:
LGTM, thank you!